src/docs/asciidoc/reference/architecture/code-structure.adoc

```diff
@@ -1,2 +1,150 @@
 :relative-path: ../../
 include::{docdir}/variables.adoc[]
+
+The purpose of this section is to explain the various elements and the interaction between them of a *ZetaPush application*.
+
+In the first instance we will see the <<reference.adoc#architecture_organization_application, organization of a ZetaPush application>>, then we will see the <<reference.adoc#architecture_files_tree, role of each file or folder>>.
+
+[#architecture_organization_application]
+== Organization of an application
+
+A *ZetaPush application* includes several components.
+You can see this with the following picture that represents the files tree of a basic application:
+
+image::{images-dir}architecture/files-tree.png[align="center"]
+
+
+The `worker` folder contains your business logic (i.e. your back code) and the `front` folder contains your front code.
+The rest of the files are needed for the functioning of the application. Each role of this files is explained below :
+
+image::{images-dir}architecture/organization-application.png[align="center"]
+<1> A ZetaPush application is mainly based on the `JavaScript ecosystem`. An application is similar to a `JavaSscript module`. So at the root of the application, we have a `package.json` file, <<reference#role_package_json, more information>> about his role is below.
+<2> In addition of the `JavaScript ecosystem`, a ZetaPush application uses the `TypeScript ecosystem` by default. So a `tsconfig.json` file is necessary to the root folder. You can see <<reference#role_tsconfig, more information about his role>> also see <<reference#javascript_vs_typescript, why we use TypeScript>>.
+<3> The `worker` folder contains your business logic.
+<4> The `front` folder contains the front part of your application.
+<5> The `application.json` file lets you configure the logic of your application. (Token, API key, Mailjet configuration, etc...).
+<6> The `.zetarc` file contains the application and account informations (developer login, developer password, appName, etc...). It should not be commited (it contains login and password) so by default it is added to the `.gitignore` file. 
```

Comment:
and account information~s~ -> and **ZetaPush** account information

No **s** for information